crates/bevy_ecs/src/schedule/condition.rs

```diff
@@ -748,10 +748,28 @@ pub mod common_conditions {
     /// app.run(&mut world);
     /// assert_eq!(world.resource::<Counter>().0, 0);
     /// ```
-    pub fn in_state<S: States>(state: S) -> impl FnMut(Res<State<S>>) -> bool + Clone {
-        move |current_state: Res<State<S>>| *current_state == state
+    pub fn in_state<S: States>(
+        state: S,
+    ) -> impl FnMut(Local<bool>, Option<Res<State<S>>>) -> bool + Clone {
+        move |warned: Local<bool>, current_state: Option<Res<State<S>>>| match current_state {
```

Comment:
We can use https://github.com/bevyengine/bevy/pull/10808 to debounce this for us :)

Reply:
Unfortunately we can't - `bevy_log` depends on `bevy_ecs` - and since states are part of `bevy_ecs` that would cause a circular dependency.

I could work on moving that from `bevy_log` to `bevy_utils` - but I don't know if it's worth having that be part of this particular PR, and worry about placing more dependencies on it this one, since it's so small & relatively simple I wouldn't want it to get delayed.

Reply:
@alice-i-cecile - created another PR for that, and I can merge that into this branch as well if that'll make things easier: https://github.com/bevyengine/bevy/pull/11419

Reply:
Yeah merge that in to make the worflow a bit easier please :)

Reply:
Done